pygmt/src/grdtrack.py

```diff
@@ -286,23 +286,35 @@ def grdtrack(points, grid, newcolname=None, outfile=None, **kwargs):
     if hasattr(points, "columns") and newcolname is None:
         raise GMTInvalidInput("Please pass in a str to 'newcolname'")
 
+    if points is None and kwargs.get("E") is None:
+        raise GMTInvalidInput("Must give 'points' or set 'profile'.")
```

Comment:
Need a unit test to cover the GMTInvalidInput case, and another one to ensure that the `profile` parameter works.

Reply:
Done in 78d98ad6e7781edfbf6e6a0eb180364fd668533e.